src/Sylius/Bundle/CoreBundle/EventListener/SimpleProductLockingListener.php

```diff
@@ -46,7 +42,7 @@ public function lock(GenericEvent $event): void
 
         if ($product->isSimple()) {
             /** @var ProductVariantInterface $productVariant */
-            $productVariant = $this->variantResolver->getVariant($product);
```

Comment:
Because now the implementation of `ProductVariantResolverInterface` will return the first **enabled** variant. Not the first variant.
It's kind an edge case but I thought that here, for a locking purpose, doesn't matter if the first variant is enabled or not. You only want the first variant to be locked so I removed the variant resolver in favor of `$product->getVariants()->first();`. Am I wrong?

Reply:
Makes sense

Reply:
In my opinion, `ProductVariantResolver` is a service to give us some product variant that we can operate on. I guess it doesn't make much sense to lock a disabled product variant (since we can't buy it in the first place). Moreover, one might have customised their product variant resolver and this part of code assumes it's not customised.

@mmenozzi @lchrusciel what do you think?

Reply:
I think we are all on the same page now. As a conclusion, it should be brought back, am I right?

Reply:
Ohh ok sorry guys. Now I understand. So I'm going to revert to its original implementation.

Reply:
Mmmm sorry guys, just one more question. Should we handle the case where there's no enabled variant for the product? And if so, how?

Reply:
Hi @lchrusciel @pamil, I decided to proceed with the revert because given the changes in the `SimpleProductSubscriber ` here https://github.com/Sylius/Sylius/pull/11241/commits/39709bd6aa119dcf5d8761e3ea7f89d123e9eb8c, for simple products the status should be the same between product and variant (at least using the admin interface).

Reply:
@pamil @lchrusciel, unfortunately we have failing Behat scenarios using that the variant resolver in the `SimpleProductLockingListener` (https://travis-ci.org/github/Sylius/Sylius/jobs/687327922#L3096). I think that it's because as I supposed before, while disabling or enabling a simple product, the variant is disabled and the locking listener has no variant to lock...

Reply:
@pamil @lchrusciel I force pushed the branch without the revert because of failing tests (https://travis-ci.org/github/Sylius/Sylius/jobs/687327922#L3096).
Maybe it's because I don't understand the purpose of this SimpleProductLockingListener but I see that it works only on simple products so only on products that have only one variant. Sounds "strange" to use a variant resolver for products that have only one variant, there's no need to resolve between many variants. We should only get that variant and lock it.

Moreover, given the fact that this PR changes the variant resolver to return the first enabled variant this may result in no variant returned if it's disabled. So if we want to use the variant resolver here we should also improve this listener to support the case where there's no enabled variant. But how? And most important, why? What's the reason to do this?

Reply:
Indeed it makes sense for simple products to get the first variant (cause that's the only one).

Reply:
Ok happy to hear that!
I'm going to resolve this conversation then.